File: erlang/scrabble-score/src/scrabble_score.erl
-module(scrabble_score).
-export([test_version/0, score/1]).

test_version() -> 1.

score([]) -> 0;
score(Word) -> lists:sum(lists:map(fun lettervalue/1, string:to_lower(Word))).

lettervalue($a) -> 1;
lettervalue($b) -> 3;
lettervalue($c) -> 3;
lettervalue($d) -> 2;
lettervalue($e) -> 1;
lettervalue($f) -> 4;
lettervalue($g) -> 2;
lettervalue($h) -> 4;
lettervalue($i) -> 1;
lettervalue($j) -> 8;
lettervalue($k) -> 5;
lettervalue($l) -> 1;
lettervalue($m) -> 3;
lettervalue($n) -> 1;
lettervalue($o) -> 1;
lettervalue($p) -> 3;
lettervalue($q) -> 10;
lettervalue($r) -> 1;
lettervalue($s) -> 1;
lettervalue($t) -> 1;
lettervalue($u) -> 1;
lettervalue($v) -> 4;
lettervalue($w) -> 4;
lettervalue($x) -> 8;
lettervalue($y) -> 4;
lettervalue($z) -> 10.

File: csharp/scrabble-score/ScrabbleScore.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ScrabbleScore
{
// Letter                           Value
// A, E, I, O, U, L, N, R, S, T       1
// D, G                               2
// B, C, M, P                         3
// F, H, V, W, Y                      4
// K                                  5
// J, X                               8
// Q, Z                               10
    private static IDictionary<char, int> scores = new Dictionary<char, int>() {
        {'a', 1}, {'b', 3}, {'c', 3}, {'d', 2}, {'e', 1}, {'f', 4}, {'g', 2},
        {'h', 4}, {'i', 1}, {'j', 8}, {'k', 5}, {'l', 1}, {'m', 3}, {'n', 1},
        {'o', 1}, {'p', 3}, {'q', 10}, {'r', 1}, {'s', 1}, {'t', 1}, {'u', 1},
        {'v', 4}, {'w', 4}, {'x', 8}, {'y', 4}, {'z', 10}, 
    };

    public static int Score(string input)
    {
        return input.ToLower().Select(x => scores[x]).Sum();
    }
}
